README.md

```diff
@@ -200,21 +200,23 @@ need different templates for different distributions.
 If some tasks need to be parameterized according to distribution and version (name of packages,
 configuration file paths, names of services), use this in the beginning of your `tasks/main.yml`:
 ```yaml
-- name: Set version specific variables
-  include_vars: "{{ item }}"
-  with_first_found:
-    - "{{ ansible_distribution }}_{{ ansible_distribution_version }}.yml"
-    - "{{ ansible_distribution }}_{{ ansible_distribution_major_version }}.yml"
-    - "{{ ansible_distribution }}.yml"
-    - "{{ ansible_os_family }}.yml"
+- name: Set distribution specific variables
+  include_vars: "{{ lookup('first_found', files) }}"
+  vars:
+    files:
+      - "{{ ansible_facts.distribution }}_{{ ansible_facts.distribution_version }}.yml"
```

Comment:
Why are you changing to `ansible_facts.*` notation?

Reply:
> Why are you changing to `ansible_facts.*` notation?

Is there a link for this?  For example - https://docs.ansible.com/ansible/latest/user_guide/playbooks_best_practices.html#operating-system-and-distribution-variance - this uses `ansible_facts['distribution']` in the example, but I can't find anywhere which says using this is better than using `ansible_distribution`, or that `ansible_distribution` etc. are deprecated.

Reply:
I am quite sure we discussed some time ago to recommend not using `ansible_facts['foo']` when there is `ansible_foo`, but I can't find it, not even in drafts. @t-woerner do you remember? I think it was you who proposed this policy.

Reply:
According to @maxamillion `ansible_facts['foo']` (or `ansible_facts.foo`) is actually preferred. `ansible_foo` relies on fact injection, which can break if there is already a fact of that name.

Reply:
So I'm not about doing things that allow people to pollute the "ansible_" prefixed setting. If a user is doing that for discovered facts, then they're already in violating of best practices.

However, this quote from the Ansible docs makes me lean towards using ansible_facts.foo/ansible_facts['foo'].

> Facts are information derived from speaking with your remote systems. You can find a complete set under the ansible_facts variable, most facts are also ‘injected’ as top level variables preserving the ansible_ prefix, but some are dropped due to conflicts. This can be disabled via the INJECT_FACTS_AS_VARS setting.

This tells me that a user can, if they wish, disable the ansible_foo options. And that would break a LOT of things. So I'm inclined to go forward with encouraging the use of the ansible_facts dictionary syntax.

Reply:
@greg-hellings I think the problem is more that there could be already a built-in Ansible variable of the same name as the fact, but I am not sure. `INJECT_FACTS_AS_VARS` is certainly also a pertinent reason.

Reply:
> @greg-hellings I think the problem is more that there could be already a built-in Ansible variable of the same name as the fact, but I am not sure. `INJECT_FACTS_AS_VARS` is certainly also a pertinent reason.

This ^^^^ :+1:

Reply:
In the future, we would like to change the default of `INJECT_FACTS_AS_VARS` to `False` mainly because fact injection has resulted in a number of CVE's over the years that we've taken many steps to mitigate. We don't have a timeline for changing the default since it will break a lot of things (though it can be easily reenabled).

Using the `ansible_facts` dict in documentation is one way we are trying to make that future change less painful for people.

Reply:
@samdoran thanks - I think https://github.com/oasis-roles/meta_standards/pull/23 is now a superset of this PR - I think I have incorporated the new and different items from this PR into #23 - if you agree, please review that PR and we can close this one